pulumi/infra/api_gateway.py

```diff
@@ -0,0 +1,112 @@
+import json
+from typing import List, Optional
+
+import pulumi_aws as aws
+
+import pulumi
+
+API_GATEWAY_LOGS_JSON_FORMAT = json.dumps(
+    {
+        # This is verbatim what you get from clicking the JSON button at
+        # https://console.aws.amazon.com/apigateway/main/monitor/logging/edit
+        "requestId": "$context.requestId",
+        "ip": "$context.identity.sourceIp",
+        "requestTime": "$context.requestTime",
+        "httpMethod": "$context.httpMethod",
+        "routeKey": "$context.routeKey",
+        "status": "$context.status",
+        "protocol": "$context.protocol",
+        "responseLength": "$context.responseLength",
+        # And some extra- integration-specific ones.
+        "integrationError": "$context.integration.error",
+        "integrationStatus": "$context.integration.status",
+        "responseType": "$context.error.responseType",
+        "message": "$context.error.message",
+    }
+)
+
+
+class ApiGateway(pulumi.ComponentResource):
+    def __init__(
+        self,
+        name: str,
+        nomad_agents_alb_security_group: pulumi.Input[str],
+        nomad_agents_alb_listener_arn: pulumi.Input[str],
+        private_subnet_ids: pulumi.Input[List[str]],
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+        """
+        Quick diatribe on how this is all hooked up:
+        Incoming request
+        --> API Gateway Route (snapshotted as a Stage)
+        --> API Gateway "integration"
+        --> VPCLink
+          (everything beyond here is defined in `platform-infrastructure`)
+        --> Nomad Agents ALB Listener (listen on :80, forward to ALB)
+        --> Nomad Agents ALB
+        --> ALB Target Group's :1234
+        --> a real Nomad Agent instance :1234
+          (everything beyond here is defined in Nomad job files)
+        --> Nomad ingress gateway (`grapl-ingress.nomad`)
+        --> `web-ui` service
+        """
+        super().__init__("grapl:ApiGateway", name, None, opts)
+
+        api = aws.apigatewayv2.Api(
+            "api",
```

Comment:
A best practice for naming resources inside a component resource (and a pattern we use pretty much everywhere in our Pulumi code --- if we're not, we should) is to incorporate the name of the component resource in it, so this would be something like `f"{name}-api"`.

(This applies to all the other resources defined here.)